### PR TITLE
grid: Graft prototype onto API

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
- Refactors state management and API calls to match #5176 
- Suspension, installation, deletion, revival, updating all work, with correct state handling
- Adds bindings for hood, docket, and treaty to @urbit/api
- Adds support for non-glob bundles
- Scraps provider names and statuses, these should come from contact-store
- Updates mock data to match new datastructures

### CODE REVIEW

File: pkg/npm/api/hood/types.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
+
+/**
+ * A pending commit, awaiting a future kelvin version
+ */
+interface Woof {
+  aeon: number;
+  weft: Weft;
+}
+
+interface Rein {
```
Comment (belisarius222) [inline pkg/npm/api/hood/types.ts:10]: FYI I recently added a `liv=?` field to `$rein`, to model the "whole desk is suspended" state.

File: pkg/grid/src/nav/Nav.tsx [hunk 2]
```diff
@@ -47,6 +47,33 @@ export type MenuState =
...
+
+  if (parts[parts.length - 1] === 'leap') {
+    parts.push('search');
+  }
+  if (parts[parts.length - 2] === 'apps') {
+    parts.pop();
+  }
+  return parts.join('/');
+}
+
```
Comment (arthyn) [inline pkg/grid/src/nav/Nav.tsx:76]: I think this is old and can be removed

File: pkg/grid/src/nav/search/Home.tsx [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+      push(createNextPath(path, input.trim()));
+    }, 300),
+    [path]
+  );
+
+  useEffect(() => {
+    if (searchInput) {
+      handleSearch(searchInput);
+    }
+  }, [searchInput]);
```
Comment (arthyn) [inline pkg/grid/src/nav/search/Home.tsx:24]: I think this is old and can be removed

File: pkg/grid/src/state/api.ts [hunk 4]
```diff
@@ -9,12 +9,13 @@ declare global {
...
+        poke: async () => {},
+        subscribe: async () => {},
+        subscribeOnce: async () => {},
+        ship: '',
+        scry: async () => {}
+      } as unknown as Urbit
     : new Urbit('', '');
 
-api.ship = window.ship;
+api.ship = 'marzod';
```
Comment (arthyn) [inline pkg/grid/src/state/api.ts:20]: this might have been left on accident

**Review by arthyn — CHANGES_REQUESTED** [review]
I like a lot of these changes. Just a few nits, but looks good.

File: pkg/npm/api/docket/types.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  image?: string;
+}
+
+export interface Charge extends Docket {
+  chad: Chad;
+}
+
+export type Chad = HungChad | GlobChad | SiteChad | InstallChad | SuspendChad;
+
+export interface HungChad {
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/npm/api/docket/types.ts:33]: https://en.wikipedia.org/wiki/Chad_(paper)

File: pkg/grid/src/state/api.ts (hunk 4 above)
Comment (liam-fitzgerald) [reply pkg/grid/src/state/api.ts:20]: Yeah I'm struggling to get vite to import the `session`.js` from the server, any ideas are welcome

File: pkg/npm/api/docket/types.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (arthyn) [reply pkg/npm/api/docket/types.ts:33]: Yeah I know, just thought it was funny

**Pull request merged**